### PR TITLE
[WIP] Adding pipelines examples

### PR DESCRIPTION
**Pull request by annapasca**
Continue #30 in order to organize the examples properly so that they can be built using sphinx gallery ...

### CODE REVIEW

File: examples/plot_power.py [hunk 1]
```diff
@@ -1,86 +1,120 @@
...
+# the values in the infosource node
 
-    return datasource
+template_path = '*%s/%s/meg/%s*rest*ica.fif'
+template_args = [['subject_id', 'session_id', 'subject_id']]
+datasource = create_datagrabber(data_path, template_path, template_args)
 
+###############################################################################
+# Ephypype creates for us a pipeline which can be connected to these
+# nodes we created. To instantiate the preprocessing node, we import it
```
Comment (jasmainak) [inline examples/plot_power.py]: preprocessing node? You mean power workflow?

File: examples/plot_power.py [hunk 2]
```diff
@@ -1,86 +1,120 @@
...
+###############################################################################
+# Finally, we are now ready to execute our workflow.
 
-    main_workflow.write_graph(graph2use='colored')  # colored
-    main_workflow.config['execution'] = {'remove_unnecessary_outputs': 'false'}
+main_workflow.config['execution'] = {'remove_unnecessary_outputs': 'false'}
 
-    main_workflow.run(plugin='MultiProc', plugin_args={'n_procs': 3})
+# Run workflow locally on 3 CPUs
+main_workflow.run(plugin='MultiProc', plugin_args={'n_procs': 3})
```
Comment (jasmainak) [inline examples/plot_power.py]: we should probably do `'n_procs': 1` by default. Not everyone has 3 cores on their laptops.

**Comment by jasmainak** [comment]
@annapasca looks good other than my nitpicks

File: examples/plot_power.py (hunk 1 above)
Comment (annapasca) [reply examples/plot_power.py]: oh yes! I changed almost everywhere :-)

File: examples/plot_power.py (hunk 2 above)
Comment (annapasca) [reply examples/plot_power.py]: ok!

**Comment by annapasca** [comment]
> @annapasca looks good other than my nitpicks

I like your nitpick style!

**Comment by annapasca** [comment]
appveyor gave this error @dmalt 

screenshot from 2018-09-28 15-17-42

**Comment by dmalt** [comment]
Yes, that's happening because nipype tries to import pwd module which is available only for, UNIX systems, when running a workflow. The reason why you see `Could not import plugin module: nipype.pipeline.plugins` instead of import error for pwd module is this exception handling from nipype/pipeline/engine/workflows.py (lines 565-579):
```
if plugin is None:
            plugin = config.get('execution', 'plugin')
        if not isinstance(plugin, (str, bytes)):
            runner = plugin
        else:
            name = '.'.join(__name__.split('.')[:-2] + ['plugins'])
            try:
                __import__(name)
            except ImportError:
                msg = 'Could not import plugin module: %s' % name
                logger.error(msg)
                raise ImportError(msg)
            else:
                plugin_mod = getattr(sys.modules[name], '%sPlugin' % plugin)
                runner = plugin_mod(plugin_args=plugin_args)
```

So if you try to run `__import__('nipype.pipeline.plugins')`  on windows machine you will get this:
```
In [1]: __import__('nipype.pipeline.plugins')
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-1-d6781e074529> in <module>()
----> 1 __import__('nipype.pipeline.plugins')

~\Miniconda3\lib\site-packages\nipype\pipeline\plugins\__init__.py in <module>()
      9 from .pbs import PBSPlugin
     10 from .oar import OARPlugin
---> 11 from .sge import SGEPlugin
     12 from .condor import CondorPlugin
     13 from .dagman import CondorDAGManPlugin

~\Miniconda3\lib\site-packages\nipype\pipeline\plugins\sge.py in <module>()
      8
      9 import os
---> 10 import pwd
     11 import re
     12 import subprocess

ModuleNotFoundError: No module named 'pwd'
```

**Comment by jasmainak** [comment]
@annapasca merged this pull request so we can fix the `appveyor` and python 2 issue in another PR

**Pull request merged**